Update setting-default-values-with-short-circuiting.md (#2035)
enkidevs/curriculum: javascript/core/functions-i/setting-default-values-with-short-circuiting.md

Instead of setting default values as below:

```javascript
function title(t) {
 if (!t) {
   t = "Untitled";
 }
}
```

We can use:

```javascript
function title(t) {
  t = t || "Untitled";
}
```

> The above will return `"Untitled"` for `t` if the provided value for `t` is falsy[1].

Author: ```suggestion
The above will return `"Untitled"` for `t` if the provided value for `t` is falsy
```

Sometimes this isn't desired. If an empty title is allowed, we would (incorrectly) get `"Untitled"` instead (because an empty string `''` is falsy).

> A more correct approach could be to use default parameters to short-circuit the default value only if the provided one is `undefined`.

Author: ```suggestion
A better approach would be to use default parameters to use the default value only if the provided one is `undefined`.
```

```javascript
function title(t = "Untitled") {
  
}
```

> If we wanted additional safety against both `undefined` and `null` (but allow all other values), we could use the nullish coalescing operator:

Author: ```suggestion
If we wanted additional safety against both `undefined` and `null` (but allow all other values), we'd use the nullish coalescing operator `??`:
```
